### PR TITLE
Added the creation of the aws_route53_record to the Custom Cognito Domain example

### PR DESCRIPTION
**Pull request by tzahari**
This is for clarification of the `aws_cognito_user_pool_domain` resource and how to generate the correct ALIAS `aws_route53_record`.
This belongs to https://github.com/hashicorp/terraform/issues/22860

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Relates #22860

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
...
```

### CODE REVIEW

File: website/docs/r/cognito_user_pool_domain.markdown [hunk 1]
```diff
@@ -34,9 +34,23 @@ resource "aws_cognito_user_pool_domain" "main" {
...
   name = "example-pool"
 }
-```
 
+data "aws_route53_zone" "example" {
+  name = "example.com"
+}
 
+resource "aws_route53_record" "auth-cognito-A" {
+  name    = "${aws_cognito_user_pool_domain.domain}"
```
Comment (jeffrafter) [inline website/docs/r/cognito_user_pool_domain.markdown]: I think this is missing the name of the custom domain:

```suggestion
  name    = "${aws_cognito_user_pool_domain.main.domain}"
```

File: website/docs/r/cognito_user_pool_domain.markdown [hunk 2]
```diff
@@ -34,9 +34,23 @@ resource "aws_cognito_user_pool_domain" "main" {
...
+  type    = "A"
+  zone_id = "${data.aws_route53_zone.example.id}"
```
Comment (jeffrafter) [inline website/docs/r/cognito_user_pool_domain.markdown]: I often see `data` used in this way. For me, I needed to change this:

```suggestion
  zone_id = "${aws_route53_zone.example.zone_id}"
```

**Comment by jeffrafter** [comment]
Just wanted to say thanks @tzahari; without the docs here I would have been stuck for quite a while!

File: website/docs/r/cognito_user_pool_domain.markdown [hunk 3]
```diff
@@ -34,9 +34,23 @@ resource "aws_cognito_user_pool_domain" "main" {
...
-```
 
+data "aws_route53_zone" "example" {
+  name = "example.com"
+}
 
+resource "aws_route53_record" "auth-cognito-A" {
+  name    = "${aws_cognito_user_pool_domain.main.domain}"
+  type    = "A"
+  zone_id = "${aws_route53_zone.example.zone_id}"
```
Comment (twelvelabs) [inline website/docs/r/cognito_user_pool_domain.markdown]: I think this needs to be `data.aws_route53_zone.example.zone_id`, or at least it needed to be for me when using terraform 0.12

**Comment by twelvelabs** [comment]
> without the docs here I would have been stuck for quite a while!

Same. Super helpful. Thank you!

File: website/docs/r/cognito_user_pool_domain.markdown (hunk 2 above)
Comment (tzahari) [reply website/docs/r/cognito_user_pool_domain.markdown]: I detected that I was adding data.

> I often see `data` used in this way. For me, I needed to change this:

I need to revert this. Cause in the lines above I referenced data.aws_route53_zone...

**Comment by fredericvl** [comment]
This is a life saver!
Please merge it to the documentation to help others.

**Pull request merged**